sql/core/src/test/scala/org/apache/spark/sql/connector/AlterTableTests.scala

```diff
@@ -1073,4 +1073,20 @@ trait AlterTableTests extends SharedSparkSession {
       assert(updated.properties === withDefaultOwnership(Map("provider" -> v2Format)).asJava)
     }
   }
+
+  test("AlterTable: replace columns") {
+    val t = s"${catalogAndNamespace}table_name"
+    withTable(t) {
+      sql(s"CREATE TABLE $t (col1 int, col2 int) USING $v2Format")
+      sql(s"ALTER TABLE $t REPLACE COLUMNS " +
+        "(col2 string COMMENT 'comment2', col3 int COMMENT 'comment3')")
```

Comment:
One question: if the `col2` already has comment but we don't specify new comment in `REPLACE COLUMNS`, will we retain the comment?

Reply:
The behavior of `REPLACE COLUMNS` is to drop all the existing columns first then add new columns. Thus, the comment will not be retained. I will update the test to reflect this.